File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Helpers/FareHelper.cs
namespace WheelShare_dotNet.Helpers
{
    public class FareHelper
    {

        private readonly IConfiguration _config;

        public FareHelper(IConfiguration config)
        {
            _config = config;
        }

        public decimal CalculateEstimatedFare(double distanceKm)
        {
            decimal baseFare = _config.GetValue<decimal>("FareConfig:BaseFare");
            decimal perKmRate = _config.GetValue<decimal>("FareConfig:PerKmRate");

            return baseFare + (decimal)distanceKm * perKmRate;
        }

        public decimal CalculateDriverEarning(decimal estimatedFare)
        {
            decimal commissionPercent =
                _config.GetValue<decimal>("CommissionConfig:DriverCommissionPercent");

            decimal commission = (commissionPercent / 100) * estimatedFare;

            return estimatedFare - commission;
        }

    }
}

File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/helper/FareHelper.java
package com.wheelshare.cabbooking.helper;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import java.math.BigDecimal;

@Component
public class FareHelper {

    @Value("${fare.base-fare:50}")
    private BigDecimal baseFare;

    @Value("${fare.per-km-rate:15}")
    private BigDecimal perKmRate;

    @Value("${commission.driver-percent:10}")
    private BigDecimal commissionPercent;

    public BigDecimal calculateEstimatedFare(double distanceKm) {
        return baseFare.add(perKmRate.multiply(BigDecimal.valueOf(distanceKm)));
    }

    public BigDecimal calculateDriverEarning(BigDecimal estimatedFare) {
        BigDecimal commission = estimatedFare.multiply(commissionPercent).divide(BigDecimal.valueOf(100));
        return estimatedFare.subtract(commission);
    }
}
